app/main/validators.py

```diff
@@ -226,3 +227,23 @@ def __call__(self, form, field):
                         raise StopValidation("Your file contains a virus")
                 finally:
                     field.data.seek(0)
+
+
+class Only2DecimalPlaces:
+    regex = re.compile(r"^[0-9]+(?:\.[0-9]{1,2})?$")
+
+    def __init__(self, message="Enter a value with 2 decimal places."):
+        self.message = message
+
+    def __call__(self, form, field):
+        if field.data and not re.match(self.regex, str(field.data)):
+            raise ValidationError(self.message)
+
+
+class IsPostcode:
```

Comment:
Why are you using the postcode regex instead of just calling `postcodes_uk.validate`?  

The below code seems a lot simpler
```
    def __call__(self, form, field):
        if field.data and not postcodes_uk.validate(str(field.data)):
            raise ValidationError(self.message)
```

Ah, I see, you used a different postcode validator to the one I suggested which doesn't have a validate function.  Comparing the two libraries only one of them still exists on github. https://github.com/GouthamSiddhaarth/postcode_validator vs https://github.com/marco-cardoso/postcodes_uk.  Ultimately I'll leave it up to you.

Reply:
I used the regex because with the validation it throws its own error message that I'd need to overwrite. Postcodes_uk had conflicting dependencies, it required an earlier version of pytest than we currently have.